### PR TITLE
:book: Update CAEP template with requirements and security model

### LINKED ISSUE #3062: Add security section to CAEP template
We've started to brush up against security concerns, particularly RBAC in the development of features and proposals. Wonder if it's worth creating an explicit section about in the CAEP template.

Initial thing could be:

### Security Model Changes

In this section, describe how Cluster API's security model would be affected by your proposal/feature.

#### Roles

Identify the expected users or systems as roles that would interact with the Kubernetes management and/or workload cluster as it relates toh 

### RBAC
RBAC (role-based access control) is the standard used for Kubernetes
authorization. This allows users to configure who can perform actions on
resources in specific scopes. RBAC can be used to enable each of the roles
defined above. In most cases, it will be desirable to have all resources be
readable by most roles, so instead we'll focus on write access for this model.

##### Read Permissions
For each role and resource pair, state whether or not the role would be able to read the resource.

| Role   | Resource 1 | Resource 2 |
|--------|------------|------------|
| Role 1 | Yes/No     | Yes/No     |
| Role 2 | Yes/No     | Yes/No     |

##### Write Permissions
For each role and resource pair, state whether or not the role would be able to write the resource.

| Role   | Resource 1 | Resource 2 |
|--------|------------|------------|
| Role 1 | Yes/No     | Yes/No     |
| Role 2 | Yes/No     | Yes/No     |

Add further commentary as needed

### PR DESCRIPTION
**Pull request by randomvariable**
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

**What this PR does / why we need it**:
Adds a security model section as well as an optional "requirements" section, which was used in https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/proposal/20200506-single-controller-multitenancy.md

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3062

### CODE REVIEW

File: docs/proposals/YYYYMMDD-template.md [hunk 1]
```diff
@@ -122,13 +129,73 @@ place your `.plantuml` files under `images/` and run `make diagrams` from the do
...
+| Role 2 | Yes   | Yes   |
+| Role 3 | No    | Yes   |
+
+##### Write Permissions
+
+|        | CRD 1 | CRD 2 |
+| ------ | ----- | ----- |
+| Role 1 | Yes   | Yes   |
+| Role 2 | Yes   | Yes   |
+| Role 3 | No    | Yes   |
```
Comment (vincepri) [inline docs/proposals/YYYYMMDD-template.md:197]: Let's remove this part? Often times permissions are changed during implementation and the proposal gets outdated quickly, the most important part is above, where we require the author(s) to think about the security model and its implications. What do you think?

File: docs/proposals/YYYYMMDD-template.md [hunk 2]
```diff
@@ -122,13 +129,73 @@ place your `.plantuml` files under `images/` and run `make diagrams` from the do
...
+##### NFR2
+
 ### Implementation Details/Notes/Constraints
 
 - What are some important details that didn't come across above.
 - What are the caveats to the implementation?
 - Go in to as much detail as necessary here.
 - Talk about core concepts and how they releate.
 
+### Security Model
```
[the remark this answers is not in the document]
Comment (randomvariable) [reply docs/proposals/YYYYMMDD-template.md:165]: Yes

File: docs/proposals/YYYYMMDD-template.md [hunk 3]
```diff
@@ -122,13 +129,73 @@ place your `.plantuml` files under `images/` and run `make diagrams` from the do
 
 #### Story 2
 
+### Requirements (Optional)
```
[the remark this answers is not in the document]
Comment (randomvariable) [reply docs/proposals/YYYYMMDD-template.md:132]: I agree on the need for clarification between the two. Goals should be more high-level whereas requirements can potentially get down in the weeds. Will expand upon, thx.

**Comment by vincepri** [comment]
@randomvariable Are you still working on this? It'd be great to merge it before we get to v1alpha4 planning phase

File: docs/proposals/YYYYMMDD-template.md (hunk 1 above)
Comment (randomvariable) [reply docs/proposals/YYYYMMDD-template.md]: Done

File: docs/proposals/YYYYMMDD-template.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (randomvariable) [reply docs/proposals/YYYYMMDD-template.md:128]: Added clarification.

**Comment by randomvariable** [comment]
@vincepri , updated.

File: docs/proposals/YYYYMMDD-template.md [hunk 4]
```diff
@@ -122,13 +125,54 @@ place your `.plantuml` files under `images/` and run `make diagrams` from the do
...
+
+The FR and NFR notation is intended to be used as cross-references across a CAEP.
+
+The difference between goals and requirements is that between an executive summary
+and the body of a document. Each requirement should be in support of a goal,
+but narrowly scoped in a way that is verifiable or ideally - testable.
+
+#### Functional Requirements
+
+Functional requirements are the properties that this design should or not include.
```
Comment (fabriziopandini) [inline docs/proposals/YYYYMMDD-template.md:143]: nit
```suggestion
Functional requirements are the properties that this design should include.
```

**Pull request merged**